File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxA = 0;
        int left = 0;
        int right = height.size() - 1;
        while (left < right)
        {
            maxA = max(maxA, (right-left)* min(height[right], height[left]));
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return maxA;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = 0;
        int left = 0;
        int right = height.length - 1;

        while (right >= 0) {
            int temp = (right-left) * Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, temp);

            if (left < height.length && height[left] < height[right]) {
                left++;
            }
            else {
                right--;
            }

            System.out.println(left + " " + right + " " + maxArea);
        }

        return maxArea;
    }
}